modules/router/src/services/fees.ts

```diff
@@ -334,199 +337,210 @@ export const calculateFeeAmount = async (
+      fromChannelFee =
+        fromChannelCode.getValue() === "0x"
+          ? GAS_ESTIMATES.createChannel.add(SIMPLE_WITHDRAWAL_GAS_ESTIMATE)
+          : SIMPLE_WITHDRAWAL_GAS_ESTIMATE;
+    } else if (finalFromBalance.lt(fromProfile.collateralizeThreshold)) {
+      // There will be a post-resolution sender collateralization
+      // gas estimates are participant sensitive, so this is safe to do
+      fromChannelFee =
+        participantFromChannel === "alice" && fromChannelCode.getValue() === "0x"
+          ? GAS_ESTIMATES.createChannelAndDepositAlice
+          : participantFromChannel === "alice"
+          ? GAS_ESTIMATES.depositAlice
+          : GAS_ESTIMATES.depositBob;
+    }
   }
 
   // when forwarding a transfer, the only immediate costs on the receiver-side
   // are the ones needed to properly collateralize the transfer
 
-  // there are several conditions that would effect the collateral costs
+  // there are several conditions that would affect the collateral costs
   // (1) channel has sufficient collateral: none
   // (2) participant == alice && contract not deployed: createChannelAndDeposit
   // (3) participant == alice && contract deployed: depositAlice
   // (4) participant == bob && contract not deployed: depositBob (channel does
   //     not need to be created for a deposit to be recognized offchain)
   // (5) participant == bob && contract deployed: depositBob
 
-  const participantToChannel = getParticipant(toChannel, routerPublicIdentifier);
-  if (!participantToChannel) {
-    return Result.fail(
-      new FeeError(FeeError.reasons.ChannelError, {
-        message: "Not in channel",
-        publicIdentifier: routerPublicIdentifier,
-        alice: toChannel.aliceIdentifier,
-        bob: toChannel.bobIdentifier,
-        channelAddress: toChannel.channelAddress,
-      }),
-    );
-  }
-  const routerBalance = getBalanceForAssetId(toChannel, toAssetId, participantToChannel);
-  // get the amount you would send
-  const isSwap = fromAssetId !== toAssetId || fromChannel.networkContext.chainId !== toChannel.networkContext.chainId;
-  const converted = isSwap
-    ? await getSwappedAmount(
-        amountToSend.toString(),
-        fromAssetId,
-        fromChannel.networkContext.chainId,
-        toAssetId,
-        toChannel.networkContext.chainId,
-      )
-    : Result.ok(amountToSend.toString());
-  if (converted.isError) {
-    return Result.fail(
-      new FeeError(FeeError.reasons.ConversionError, {
-        swapError: jsonifyError(converted.getError()!),
-      }),
-    );
-  }
-  if (BigNumber.from(routerBalance).gte(converted.getValue())) {
-    // channel has balance, no extra gas required to facilitate transfer
+  if (!toChannel) {
+    // if no channel exists, we need to account for the full channel deployment always
+    // NOTE: same assumption as above for alice
+    return Result.ok([fromChannelFee, GAS_ESTIMATES.createChannelAndDepositAlice]);
```

Comment:
You're not accounting for potential reclamation of the receiver channel here

Reply:
If the `toChannel` doesn't exist, there's always 0 balance in the channel so it will always be a JIT collateralization, which is for the exact amount.

Reply:
Ahhhh got it